### PR TITLE
update documentation on nullptr deallocation behavior

### PR DESCRIPTION
**Pull request by kab163**

### CODE REVIEW

**Review by davidbeckingsale — APPROVED** [review]

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
LGTM

File: src/umpire/Allocator.hpp [hunk 1]
```diff
@@ -55,10 +55,12 @@ class Allocator {
    * \brief Free the memory at ptr.
    *
    * This method will throw an umpire::Exception if ptr was not allocated
-   * using this Allocator. If you need to deallocate memory allocated by an
-   * unknown object, use the ResourceManager::deallocate method.
+   * using this Allocator. If the pointer hasn't been used (a nullptr),
```
Comment (mcfadden8) [inline src/umpire/Allocator.hpp:58]: Being a nullptr, does not really indicate whether the pointer has been used.  It simply means that the pointer was set to nullptr.  I suggest the following:

```suggestion
   * using this Allocator. If the value of the pointer is set to nullptr, 
```

File: src/umpire/Allocator.inl [hunk 2]
```diff
@@ -43,7 +43,7 @@ inline void Allocator::deallocate(void* ptr)
   UMPIRE_LOG(Debug, "(" << ptr << ")");
 
   if (!ptr) {
-    UMPIRE_LOG(Info, "Deallocating a null pointer");
+    UMPIRE_LOG(Info, "Deallocating a null pointer (This behavior is purposefully allowed!)");
```
Comment (mcfadden8) [inline src/umpire/Allocator.inl:46]:
```suggestion
    UMPIRE_LOG(Info, "Deallocating a null pointer (This behavior is intentionally allowed and ignored)");
```

File: src/umpire/Allocator.hpp (hunk 1 above)
Comment (kab163) [reply src/umpire/Allocator.hpp]: Sounds good. I was trying to get the wording just right... words are hard

**Review by mcfadden8 — APPROVED** [review]
Thanks Kristi,

This looks good to me.

File: src/umpire/Allocator.inl (hunk 2 above)
Comment (kab163) [reply src/umpire/Allocator.inl]: I think that is more clear, thanks!

**Comment by kab163** [comment]
LGTM

**Pull request merged**